crash server

Dmitriy51 opened the pull request:
if(sscanf(params, "d", player)) return SendClientMessagef(playerid, red,"|USE|: {333333} /ban |id]");
if(!SSCANF_IsConnected(player)) return SendClientMessagef(playerid, red,"|ERROR|: {00FFFF}Player not connected!");

SSCANF_IsConnected(player) this function crash the server, if (player = -11111111111)

Y-Less committed:
Starting on `[*]` length option.  Arrays done, enums more complex...

Y-Less committed:
Collect lengths first in enums.

Y-Less committed:
Correct some parameters and positions.

Y-Less committed:
Include args everywhere.

Y-Less committed:
Use `args` references.

Y-Less committed:
Update copyright by NINE YEARS.

Y-Less committed:
Include `sscanf.def`, not `YSF.def`...

Y-Less committed:
Allow `s[(32)]`.

Y-Less committed:
Update load message.

Y-Less committed:
Remove old comments.

Y-Less committed:
Improve `extract`:

`P<,>` works.
Arrays and strings use `*` for all lengths.
Int arrays fixed.

Y-Less committed:
Remove `z`!

Y-Less committed:
Reuse `z` for packed strings.

Y-Less committed:
Porting the old topic to the README.

Y-Less committed:
Remove all deprecations.  It's been 10 years...

Y-Less committed:
More documentation.

Y-Less committed:
String length error.

Y-Less committed:
Document `K` improvements.

Y-Less committed:
Fix some formatting issues.

Y-Less committed:
`OnCachedInit`.

Y-Less committed:
Fix `extract p -> i, Float:f;`

It was because this generated `#if`, which is a stringised keyword.  Fortunately a `""` prefix resolves this.  I could have switched from `i` to `d`, but that would just change the problem to `extract p -> i, oct:x` generating `#do` instead.

Y-Less committed:
Add `x` to list of all specifiers.

Y-Less committed:
Alphebatise the all specifiers table.

Y-Less committed:
Colours documentation.

Y-Less committed:
Line numbers.

Y-Less committed:
Colours.

Y-Less committed:
Tests and fixes for `M`

Y-Less committed:
Error files and lines.

Y-Less committed:
2.10.0 version bump.

Y-Less committed:
2020 v2.10.0

Y-Less committed:
Fix Linux build.

Y-Less committed:
pawn.json

Y-Less committed:
2.10.1

Y-Less committed:
Fix parameter counts.

Y-Less committed:
2.10.2

neiam00 committed:
Missing `

Y-Less committed:
Merge pull request #2 from leHeix/patch-3

Missing ` in README specifiers table

Y-Less committed:
prehooks

Y-Less committed:
Export an `sscanf` function for other plugins.

Y-Less committed:
Save/restore gFormat

Y-Less committed:
Rename to `PawnSScanf`.

Y-Less committed:
Correct README version.

Y-Less commented:
1. This isn't an issue.
2. This is the old repo.
3. Why are you even calling that function in the first place?  That's an undocumented internal function.

Y-Less committed:
Hide internal `SSCANF_IsConnected` function.

Cheaterman committed:
Fix CFLAGS/LDFLAGS on Linux

-O3 is too aggressive, -O2 means nothing when linking, and -fshort-wchar
is potentially harmful (and not otherwise useful).

# Conflicts:
#	makefile

Y-Less committed:
Add credits.

Y-Less committed:
`SSCANF_SetOption`, `SSCANF_GetOption`.

Y-Less committed:
`SSCANF_Version`.

Y-Less committed:
Overload the `SSCANF_Version` and `SSCANF_Option` functions.

Y-Less committed:
More and less function overloading.

Y-Less committed:
Documentation.

Y-Less committed:
Better BCD.

Y-Less committed:
Documentation.

Y-Less committed:
Fix up the license.

Y-Less committed:
Spacing

Y-Less committed:
Put special thanks back in.

Y-Less committed:
Oxford comma.

Cheaterman committed:
Add state _ALS : _ALS_go to SSCANF_RunInit and SSCANF_OnNPCModeInit

sscanf otherwise wouldn't work on setups that don't have any other
script making the call.

Southclaws committed:
Important typo fix

Fixes critical bug, please merge ASAP.

Y-Less committed:
Contents

Commit:
Intro and full contents.

Commit:
Promote some contents lines.

Commit:
Introduction.

Commit:
Future plans.

Commit:
Note about clobbering.

Commit:
Compilation

Commit:
Fix stringise version number.

Commit:
Allow trailing string literals.

Commit:
Bump the version number.

Commit:
`SSCANF_VERSION` macro for compile-time checks.

Commit:
Document `SSCANF_VERSION`.

Commit:
Add `SSCANF_VERSION` for compile-time checks.

Commit:
NPC init/cleanup functions.

Commit:
Shuffling files.

Commit:
Import the whole AMX runtime, instead of using remote stubs.

Commit:
Remove all references to the `pAMXFunctions` stub functions.

Commit:
Compiling against the new AMX version.

Commit:
Make copies of the AMX functions as `npcamx_Function`.

Commit:
Put `pAMXFunctions` back.

Commit:
Use passed functions where possible.

Commit:
Fix export types.

Commit:
Document NPC modes again.

Commit:
Fix NPC include.

DEntis-T committed:
Fix line endings

DEntis-T committed:
Fix line endings

Y-Less committed:
Merge pull request #7 from DEntis-T/master

Line endings - fixed

Commit:
Linux fixes.

Commit:
2.11.2

Commit:
Another 2.11.2

Commit:
Update credits.

Se8870 committed:
Rename amxplugin.cpp to amxplugin.c

Y-Less committed:
Merge pull request #8 from Se8870/master

Fix linux build

Commit:
Don't use line formatted errors when there are no lines.

Commit:
`SSCANF_Levenshtein`.

Commit:
Documentation.

Commit:
Fill out links.

Commit:
Update credits.

Commit:
Merge two PRs at once.

Commit:
Credits.

Commit:
Documentation.

Commit:
2.11.3

Commit:
Update contents.

Commit:
Update copyright year.

Commit:
Loads of documentation comments.

Commit:
2.11.4 version bump.

Commit:
`#ifdef`s for comments.

Commit:
BCD and comments.

Commit:
Was going to allow BCD encoding of `0x03023664`...

Commit:
But decided against it.

Commit:
More documentation.

Commit:
More documentation.

Commit:
More documentation.

Commit:
Add `<library>sscanf</library>`

Commit:
Annotate an odd looking check.

Commit:
Separate publics, each to a separate `#ifdef` block.

Commit:
Complete pawndoc documentation.

Commit:
`<remarks>`

Commit:
Try comment out the invalid transition XML.

Commit:
Move both `state` calls to one place.

Commit:
Fix a typo.

Y-Less committed:
Move a comment to pawndoc.

Commit:
Unify options documentation.

Commit:
Put the sscanf intro in the include.

Commit:
Compiler fallback information.

Y-Less committed:
Note about why `#LEN` doesn't always work.

Commit:
Add `pawndoc="true"`.

Commit:
Remove empty pawndoc blocks.
